ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallDialog.kt

```diff
@@ -50,15 +50,19 @@ fun PaywallDialog(
         val dismissRequest = {
             shouldDisplayDialog = false
         }
+        val paywallOptions = paywallDialogOptions.toPaywallOptions(dismissRequest)
+
+        val viewModel = getPaywallViewModel(options = paywallOptions)
 
         Dialog(
             onDismissRequest = {
                 dismissRequest()
+                viewModel.close()
```

Comment:
I would like to add tests to make sure we're calling these appropriately, but those would be UI tests, not snapshot tests, which might be a bit harder to write... Will try to write these tests after.